docs/src/developers/git.md

```diff
@@ -3,36 +3,83 @@
 Git is very flexible and this can be a bit too much at first.
 This guide provides a workflow that should allow you to get things done and not lead you in any tricky situations.
 Moreover, it guarantees that the whole team has the same setup.
-
-## Add your remote
+This guide assumes you have already followed and completed your [Set up](@ref).
+This guide assumes that your Github login is `jdupont`. Replace it by the appropriate login since that is not the case.
+
+You can be in 2 situations:
+1) You don't have write access to `origin`. If you don't know what this means, it means you are in that situation.
+   Once you have enough mastery of Git, we might give you write access but it's best to start in this situation.
+2) You have write access to `origin`. This means you can push directly to `master` which you should never do. We only give
+   you write access when we trust that you won't mess up.
+
+We have a `master` branch that contains the latest version of all **merged** changes.
+There is three `master` branches:
+* `origin/master`: that is the branch at https://github.com/dionysos-dev/Dionysos.jl, it is always the most up to date.
+* Your local `master` branch: that is the state of the branch on your computer. It may be a few commits behind `origin/master` as they do not synchronize automatically. You can update it by following [Switching to the master branch and updating it](@ref).
```

Comment:
```suggestion
* Your local `master` branch: that is the state of the branch on your computer. It may be a few commits behind `origin/master` as they do not synchronize automatically. You can update it by following [Switch to the master branch and update it](@ref).
```